helper/schema/field_reader_map.go

```diff
@@ -56,10 +56,11 @@ func (r *MapFieldReader) readMap(k string) (FieldReadResult, error) {
 	prefix := k + "."
 	r.Map.Range(func(k, v string) bool {
 		if strings.HasPrefix(k, prefix) {
+			resultSet = true
+
 			key := k[len(prefix):]
 			if key != "#" {
 				result[key] = v
-				resultSet = true
 			}
```

Comment:
This fixes another issue where MapFieldReader would not map a read map as `Exists == true` unless it had one non-`#` element.